crates/sui-core/src/consensus_manager/mysticeti_manager.rs

```diff
@@ -107,81 +102,69 @@ impl ConsensusManagerTrait for MysticetiManager {
+                epoch_store
+                    .committee()
+                    .authority_index(&name)
+                    .expect("Should have valid index for own authority") as usize,
+            )
+            .expect("Should have valid index for own authority");
 
         let registry = Registry::new_custom(Some("mysticeti_".to_string()), None).unwrap();
 
-        const MAX_RETRIES: u32 = 2;
-        let mut retries = 0;
-
-        loop {
-            let private_key = self.network_keypair.copy().private();
-
-            // TODO: that should be replaced by a metered channel. We can discuss if unbounded approach
-            // is the one we want to go with.
-            #[allow(clippy::disallowed_methods)]
-            let (commit_sender, commit_receiver) = unbounded_channel();
-
-            let consensus_handler = consensus_handler_initializer.new_consensus_handler();
-            let consumer = CommitConsumer::new(
-                commit_sender,
-                consensus_handler.last_executed_sub_dag_index().await,
-            );
-
-            match Validator::start_production(
-                authority_index,
-                committee.clone(),
-                &parameters,
-                config.clone(),
-                registry.clone(),
-                Signer(Box::new(private_key.0.clone())),
-                consumer,
-                tx_validator.clone(),
-            )
-            .await
-            {
-                Ok((validator, tx_sender)) => {
-                    let registry_id = self.registry_service.add(registry);
-
-                    self.validator
-                        .swap(Some(Arc::new((validator, registry_id))));
-
-                    // create the client to send transactions to Mysticeti and update it.
-                    self.client.set(MysticetiClient::new(tx_sender));
-
-                    // spin up the new mysticeti consensus handler to listen for committed sub dags
-                    let handler =
-                        MysticetiConsensusHandler::new(consensus_handler, commit_receiver);
-                    self.consensus_handler.store(Some(Arc::new(handler)));
-
-                    break;
-                }
-                Err(err) => {
-                    retries += 1;
-
-                    self.metrics.start_mysticeti_retries.set(retries as i64);
-
-                    if retries >= MAX_RETRIES {
-                        panic!(
-                            "Failed starting Mysticeti, maxed out retries {}: {:?}",
-                            retries, err
-                        );
-                    }
-
-                    tracing::error!("Failed starting Mysticeti, retry {}: {:?}", retries, err);
-                    tokio::time::sleep(tokio::time::Duration::from_secs(1)).await;
-                }
-            }
-        }
+        // TODO: that should be replaced by a metered channel. We can discuss if unbounded approach
+        // is the one we want to go with.
```

Comment:
👍

Reply:
I think unbounded is fine. But we should build a wrapper to meter / monitor unbounded channels too. Adding a work item.